src/rdkafka_telemetry.c

```diff
@@ -359,21 +377,24 @@ static void rd_kafka_send_push_telemetry(rd_kafka_t *rk,
                                      compressed_metrics_payload_size,
                                      rk->rk_telemetry.telemetry_max_bytes);
                 }
+
+                rd_kafka_dbg(
+                    rk, TELEMETRY, "PUSH",
+                    "Sending PushTelemetryRequest with terminating = %s",
+                    RD_STR_ToF(terminating));
+                rd_kafka_PushTelemetryRequest(
+                    rkb, &rk->rk_telemetry.client_instance_id,
+                    rk->rk_telemetry.subscription_id, terminating,
+                    compression_used, compressed_metrics_payload,
+                    compressed_metrics_payload_size, NULL, 0,
+                    RD_KAFKA_REPLYQ(rk->rk_ops, 0),
+                    rd_kafka_handle_PushTelemetry, NULL);
         } else {
-                rd_kafka_dbg(rk, TELEMETRY, "PUSH",
-                             "No metrics to push. Sending empty payload.");
+                rd_kafka_log(rk, LOG_WARNING, "PUSH",
+                             "Telemetry metrics encode error, not sending "
+                             "a NULL payload");
```

Comment:
`not sending metrics` instead of `not sending a NULL payload` is better I guess.